src/zh/zaimanhua/src/eu/kanade/tachiyomi/extension/zh/zaimanhua/Zaimanhua.kt

```diff
@@ -0,0 +1,202 @@
+        )
+        .setToken(preferences.getString("TOKEN", "")!!)
+        .build()
+
+    // Detail
+    // path: "/comic/detail/mangaId"
+    override fun mangaDetailsRequest(manga: SManga): Request =
+        GET("$apiUrl/comic/detail/${manga.url}", apiHeaders)
+
+    override fun mangaDetailsParse(response: Response): SManga = try {
+        response.parseAs<ResponseDto<ResponseDto<MangaDto>>>().data!!.data!!.toSManga()
+    } catch (_: Throwable) {
+        throw Exception("获取漫画信息失败，可能需要用户登录")
+    }
+
+    // Chapter
+    override fun chapterListRequest(manga: SManga): Request = mangaDetailsRequest(manga)
+
+    override fun chapterListParse(response: Response): List<SChapter> = try {
+        response.parseAs<ResponseDto<ResponseDto<MangaDto>>>().data!!.data!!.parseChapterList()
+    } catch (_: Throwable) {
+        throw Exception("获取漫画信息失败，可能需要用户登录")
+    }
+
+    override fun imageUrlParse(response: Response): String = throw UnsupportedOperationException()
+
+    // Pagelist
+    // path: "/comic/chapter/mangaId/chapterId"
+    override fun pageListRequest(chapter: SChapter) =
+        GET("$apiUrl/comic/chapter/${chapter.url}", apiHeaders)
+
+    override fun pageListParse(response: Response): List<Page> =
+        response.parseAs<ResponseDto<ResponseDto<ChapterImagesDto>>>().data!!.data!!.images.mapIndexed { index, it ->
+            Page(index, "", it)
+        }
+
+    // Popular
+    override fun popularMangaRequest(page: Int): Request =
+        GET("$apiUrl/comic/rank/list?tag_id=0&by_time=3&rank_type=0&page=$page")
+
+    override fun popularMangaParse(response: Response): MangasPage = latestUpdatesParse(response)
+
+    // Search
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request =
+        GET("$apiUrl/search/index?keyword=$query&source=0&page=$page&size=20", apiHeaders)
+
+    override fun searchMangaParse(response: Response): MangasPage =
+        response.parseAs<ResponseDto<PageDto>>().data!!.toMangasPage()
+
+    // Latest
+    // "$apiUrl/comic/update/list/1/$page" is same content
+    override fun latestUpdatesRequest(page: Int): Request =
+        GET("$apiUrl/comic/update/list/0/$page")
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val mangas = response.parseAs<ResponseDto<List<PageItemDto>>>().data!!
+        return MangasPage(mangas.map { it.toSManga() }, true)
+    }
+
+    override fun setupPreferenceScreen(screen: PreferenceScreen) {
+        ListPreference(screen.context).apply {
+            EditTextPreference(screen.context).apply {
+                key = "USERNAME"
+                title = "用户名"
+                setOnPreferenceChangeListener { _, newValue ->
+                    preferences.edit().putString("USERNAME", newValue as String).commit()
+                    true
+                }
```

Comment:
`setOnPreferenceChangeListener` isn't needed, app will set the preference itself